macros/PGauxiliaryFunctions.pl

```diff
@@ -106,73 +106,35 @@ sub Round {
 }
 
 #least common multiple
-#VS 6/29/2000
 # ^function lcm
 sub lcm {
-	my $a = shift;
-	my $b = shift;
-
-	#reorder such that $a is the smaller number
-	if ($a > $b) {
-		my $temp = $a;
-		$a = $b;
-		$b = $temp;
-	}
-
-	my $lcm = 0;
-	my $curr = $b;;
-
-	while($lcm == 0) {
-		$lcm = $curr if ($curr % $a == 0);
-		$curr += $b;
-	}
-
-	$lcm;
-
+        do {warn 'Cannot take lcm of the empty set'; return;} unless (@_);
+        my $a = abs(shift);
+        if ($a == 0) {return 0;}
+        return $a unless (@_);
+        my $b = abs(shift);
+        if ($b == 0) {return 0;}
+        else {return lcm($a*$b/gcf($a,$b),@_);};
 }
 
 
 # greatest common factor
-# takes in two scalar values and uses the Euclidean Algorithm to return the gcf
-#VS 6/29/2000
+# takes in scalar values and uses the Euclidean Algorithm to return the gcf
 # ^function gcf
 sub gcf {
-        my $a = abs(shift);	# absolute values because this will yield the same gcd,
-        my $b = abs(shift);	# but allows use of the mod operation
-
-	# reorder such that b is the smaller number
-	if ($a < $b) {
-		my $temp = $a;
-		$a = $b;
-		$b = $temp;
-	}
-
-	return $a if $b == 0;
-
-	my $q = int($a/$b);	# quotient
-	my $r = $a % $b;	# remainder
-
-	return $b if $r == 0;
-
-	my $tempR = $r;
-
-	while ($r != 0) {
-
-		#keep track of what $r was in the last loop, as this is the value
-		#we will want when $r is set to 0
-		$tempR = $r;
-
-		$a = $b;
-		$b = $r;
-		$q = $a/$b;
-		$r = $a % $b;
-
-	}
-
-	$tempR;
+        do {warn 'Cannot take gcf of the empty set or an all-zero set'; return;} unless (@_);
+        my $a = abs(shift);
+        if ($a == 0) {return gcf(@_);}
+        return $a unless (@_);
+        my $b = abs(shift);
+        if ($b == 0) {return gcf($a,@_);}
```

Comment:
There is no need for lines 127 and 130 above.  If $a is 0 and $a > $b (enforced by the next line below), then the while loop won't run, and the next iteration will remove off $a and the iteration after that will remove $b.  If $a > 0 and $b is 0, then the next line below will switch that and again the while loop won't run and the next iteration will remove $a.

Reply:
I see.  Line 127 gives the warning about not taking the gcf of an all-zero set on the next iteration.  So that line is needed.  Line 130 is not though.  If it gets there then $a is not zero, and the next iteration will remove $b.